axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/Scheduler.java

```diff
@@ -0,0 +1,96 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * Responsible to scheduler at fixed rate a generic task.
+ * The scheduling can be started an stopped at any time.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2
+ */
+public class Scheduler {
+
+    private final Logger log = LoggerFactory.getLogger(Scheduler.class);
+
+    private final ScheduledExecutorService scheduler;
+
+    private final long initialDelay;
+
+    private final long rate;
+
+    private final AtomicReference<ScheduledFuture> scheduledTask = new AtomicReference<>();
+
+    private final Runnable runnable;
```

Comment:
I feel `runnable` is to generic a name. You however already use `scheduledTask` for the `AtomicReference`, losing the only sensible name I can imagine for this `Runnable`...

If we can think of something better, I am in favor of renaming it.
If we cant, then lets leave it as is.

Reply:
I've used task i.o. runnable